File: Java/ByTier/A/2155_A_El_fucho.java
// Problem: CF 2155 A - El fucho
// https://codeforces.com/contest/2155/problem/A

/*
 * Problem: Tournament Matches Count
 * 
 * Description:
 * In a modified double-elimination football tournament with n teams, we need to determine
 * the total number of matches played until a single winner emerges.
 * 
 * Algorithm/Techniques:
 * - Simulates tournament rounds where teams are split into winners and losers groups.
 * - Each match eliminates exactly one team from the competition.
 * - The key insight is that in any valid elimination tournament structure, exactly (n-1) teams
 *   must be eliminated, hence (n-1) matches are needed to determine a single winner.
 * 
 * Time Complexity: O(t), where t is the number of test cases.
 * Space Complexity: O(1), constant extra space used.
 * 
 * Example:
 * For n=2: 2 matches
 * For n=3: 4 matches
 */

import java.util.Scanner;
public class Div2 {
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        int t=sc.nextInt(); // Read number of test cases
        while(t-->0){ // Process each test case
            int n=sc.nextInt(); // Read number of teams
            System.out.println(2*(n-1)); // Output total matches (derived from logic above)
        }
    }
}


// https://github.com/VaHiX/CodeForces/

File: Kotlin/ByRound/2155/2155_A_El_fucho.kt
// Problem: CF 2155 A - El fucho
// https://codeforces.com/contest/2155/problem/A

/*
 * Problem: Double Elimination Tournament
 * Purpose: Calculate total number of matches played in a double-elimination tournament with n teams.
 *
 * Algorithm/Techniques:
 * - Simulates the pairing and elimination process described in the problem.
 * - Uses two counters: 'wi' for winners group, 'lo' for losers group.
 * - In each iteration:
 *   1. Pair up teams in winners group, move losers to losers group.
 *   2. Pair up teams in losers group, eliminate losers.
 * - Continue until both groups have 1 team each, then one final match.
 *
 * Time Complexity: O(log n) per test case
 * Space Complexity: O(1)
 */

import java.util.StringTokenizer

fun solve() {
    val n = readLine()!!.toInt() // Read number of teams
    var wi = n // Initial number of teams in winners group
    var lo = 0 // Initial number of teams in losers group
    var ans = 0 // Total matches counter
    
    while (true) {
        if (wi == 1 && lo == 1) { // Both groups have one team, finalize with one match
            ans++
            break
        } else if ((wi == 1 && lo == 0) || (lo == 1 && wi == 0)) { // One group is empty, can't proceed
            break
        }
        
        val a = wi / 2 // Number of matches in winners group (a teams lose and move to losers)
        wi -= a // Update winners group size after matches
        val b = lo / 2 // Number of matches in losers group (b teams get eliminated)
        lo -= b // Update losers group size after matches
        lo += a // Add losers from winners group to losers group
        
        ans += a + b // Increment total match count
    }
    
    println(ans)
}

fun main() {
    val t = readLine()!!.toInt() // Read number of test cases
    repeat(t) {
        solve()
    }
}


// https://github.com/VaHiX/CodeForces/